### PR TITLE
Enable custom session storage client

### PR DESCRIPTION
**Pull request by innovation-stack**
# Summary
This PR enables support for custom session storage client. The `moduleConfig` has been enhanced to support custom session storage via `sessionStorageClient` property.

## Change Type

- [ ] Bug Fix
- [X] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [X] My code follows this project's style guidelines
- [X] I've reviewed my own code
- [ ] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [X] My changes generate no new warnings
- [ ] I've added tests that prove my fix is effective or my feature works
- [ ] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

# Additional Information

# Branch Naming

### CODE REVIEW

File: packages/modules/src/utils/Types.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ export interface SessionKeyManagerModuleConfig extends BaseValidationModuleConfi
   version?: ModuleVersion;
   nodeClientUrl?: string;
   smartAccountAddress: string;
-  storageType?: StorageType;
+  sessionStorageClient?: ISessionStorage;
```
Comment (livingrockrises) [inline packages/modules/src/utils/Types.ts:23]: can rename this to customSessionStorageClient

File: packages/modules/src/utils/Types.ts [hunk 2]
```diff
@@ -31,17 +32,12 @@ export interface BatchedSessionRouterModuleConfig extends BaseValidationModuleCo
   sessionManagerModuleAddress?: string;
   nodeClientUrl?: string;
   smartAccountAddress: string;
-  storageType?: StorageType;
```
Comment (livingrockrises) [inline packages/modules/src/utils/Types.ts:34]: can we not remove this storageType please

File: packages/modules/src/utils/Types.ts [hunk 3]
```diff
@@ -31,17 +32,12 @@ export interface BatchedSessionRouterModuleConfig extends BaseValidationModuleCo
...
 
   // sessionSigner?: Signer
   // sessionPubKey?: string
   // nodeClientUrl?: string
 }
 
-export enum StorageType {
```
Comment (livingrockrises) [inline packages/modules/src/utils/Types.ts:41]: can we not remove this storageType please

File: packages/modules/src/SessionKeyManagerModule.ts [hunk 4]
```diff
@@ -7,7 +7,6 @@ import {
   SessionKeyManagerModuleConfig,
   ModuleVersion,
   CreateSessionDataParams,
-  StorageType,
```
Comment (livingrockrises) [inline packages/modules/src/SessionKeyManagerModule.ts:10]: can we not remove this storageType please

File: packages/modules/src/BatchedSessionRouterModule.ts [hunk 5]
```diff
@@ -61,8 +61,7 @@ export class BatchedSessionRouterModule extends BaseValidationModule {
       const sessionModule = await SessionKeyManagerModule.create({
         moduleAddress: instance.sessionManagerModuleAddress,
         smartAccountAddress: moduleConfig.smartAccountAddress,
-        nodeClientUrl: moduleConfig.nodeClientUrl,
-        storageType: moduleConfig.storageType,
```
Comment (livingrockrises) [inline packages/modules/src/BatchedSessionRouterModule.ts:65]: can we not remove this storageType please

File: packages/modules/src/SessionKeyManagerModule.ts [hunk 6]
```diff
@@ -66,11 +65,11 @@ export class SessionKeyManagerModule extends BaseValidationModule {
     instance.nodeClient = new NodeClient({
       txServiceUrl: moduleConfig.nodeClientUrl ?? NODE_CLIENT_URL,
     });
-
-    if (!moduleConfig.storageType || moduleConfig.storageType === StorageType.LOCAL_STORAGE) {
```
Comment (livingrockrises) [inline packages/modules/src/SessionKeyManagerModule.ts:70]: can we not remove this storageType please

instead I suggest

if (moduleConfig.sessionStorageClient) {
      instance.sessionStorageClient = moduleConfig.sessionStorageClient;
    } else {
    
    // switch case here on moduleConfig.storageType if provided
    // if case if LOCAL_STORAGE then create SessionLocalStorage
    // throw in default case
    
}

**Review by livingrockrises — COMMENTED** [review]
reviewed.
make sure you use node v18 
run ./rebuild.sh in root
do yarn
yarn build

should build successfully

File: packages/modules/src/SessionKeyManagerModule.ts (hunk 4 above)
Comment (innovation-stack) [reply packages/modules/src/SessionKeyManagerModule.ts:10]: Fixed

File: packages/modules/src/utils/Types.ts (hunk 1 above)
Comment (innovation-stack) [reply packages/modules/src/utils/Types.ts:24]: Fixed

File: packages/modules/src/utils/Types.ts (hunk 2 above)
Comment (innovation-stack) [reply packages/modules/src/utils/Types.ts]: Fixed

File: packages/modules/src/utils/Types.ts (hunk 3 above)
Comment (innovation-stack) [reply packages/modules/src/utils/Types.ts]: Fixed

File: packages/modules/src/SessionKeyManagerModule.ts [hunk 7]
```diff
@@ -73,6 +73,10 @@ export class SessionKeyManagerModule extends BaseValidationModule {
       throw new Error("Invalid storage type");
     }
 
+    if (moduleConfig.customSessionStorageClient) {
```
Comment (livingrockrises) [inline packages/modules/src/SessionKeyManagerModule.ts:76]: better to move this code above as per logic I suggested in comments earlier.

**Review by livingrockrises — APPROVED** [review]
LGTM

**Review by AmanRaj1608 — APPROVED** [review]

**Pull request merged**